cmd/file/io.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
```

Comment:
Package `cmd/file` is not the `main` package and from its path it suggests that. I think that it would be more correct to utilize here internal package feature and the move this package to `cmd/internal/file`.

Reply:
Will it then still be accessible by `cmd/bee-split/main.go`

I was thinking of this structure (but left it like this now to make the diffs for split and join clearer)

```
cmd/file/io.go
cmd/file/bee-split/main.go
cmd/file/bee-join/main.go
cmd/file/bee-file/main.go
```

That still doesn't fit what you are describing as a problem though.

Reply:
I would preserve that all first subdirectories under cmd are only main packages.

Reply:
I will try with the `internal` but I was under the impression that `internal` will only work for the package in its immediate parent directory...? in that case sharing code between those binaries is not possible using that method.

Reply:
They just need to share the a common ancestor, which is here `github.com/ethersphere/bee/cmd`.